edge-util/src/Microsoft.Azure.Devices.Edge.Storage/SequentialStore.cs

```diff
@@ -97,6 +97,41 @@ public async Task<bool> RemoveFirst(Func<long, T, Task<bool>> predicate, Cancell
             }
         }
 
+        public async Task<int> RemovePercentFromHead(Func<long, T, Task> callback, int percentage, CancellationToken cancellationToken)
+        {
+            using (await this.headLockObject.LockAsync(cancellationToken))
+            {
+                long itemsToDelete = 0;
+                using (await this.tailLockObject.LockAsync(cancellationToken))
+                {
+                    // Tail offset could change here, but not holding a lock for efficiency.
```

Comment:
This comment doesn't apply here since you *are* taking the tail object lock, unlike `RemoveFirst`.